Applied Enhancements

reynoldtan opened the pull request:
- Dropped Highly Recommended texts from Experiment option in download.
- Allowed multiple selection of species in download.
- Disabled configuration option when data has been uploaded in
configuration.
- Set image 1 an 2 optional in upload.
- Disabled/hide entire mapping information fields when not applicable
in upload
and renamed the section to Related Ontology Terms.
- Changed title of description field to Description of Trait Including
Collection Method in upload.
- Number of traits described counter increments w/0 ontology
information in upload
- Added a link to phenotypes when saving data is complete in upload.

reynoldtan committed:
Fixed bug in grouping of mean values

reynoldtan committed:
FIXED BUG

- R Friendly version disables when updating options
- Germplasm field does not clear when no match found.

reynoldtan committed:
Applied Enhancements

- Dropped Highly Recommended texts from Experiment option in download.
- Allowed multiple selection of species in download.
- Disabled configuration option when data has been uploaded in
configuration.
- Set image 1 an 2 optional in upload.
- Disabled/hide entire mapping information fields when not applicable
in upload
and renamed the section to Related Ontology Terms.
- Changed title of description field to Description of Trait Including
Collection Method in upload.
- Number of traits described counter increments w/0 ontology
information in upload
- Added a link to phenotypes when saving data is complete in upload.

laceysanderson reviewed (commented):
- [x] Code Review Approved
- [ ] Functional testing (still to be done)

laceysanderson reviewed (changes requested):
- [x] Code Review Approved
- [ ] Functional Testing
    - [x] Dropped Highly Recommended texts from Experiment option in download.
    - [ ] Allowed multiple selection of species in download.
         - It's a huge usability concern that we can't even select traits once multiple species are selected. We need to be able to select any combination of filter criteria: see #25  (hold off until after this release).
         - Not to mention it's very confusing that you tell me to "fill out as many of the following filters as possible to narrow the phenotype set" but disable the form elements. 
    - [x] Disabled configuration option when data has been uploaded in configuration.
    - [x] Set image 1 and 2 optional in upload.
    - [ ] Disabled/hide entire mapping information fields when not applicable in upload and renamed the section to Related Ontology Terms.
         - Currently disabled but as a data uploader I find this confusing. Instead we should hide the "Related Ontology Terms" fieldset entirely.
    - [x] Changed title of description field to Description of Trait Including Collection Method in upload.
    - [x] Number of traits described counter increments w/0 ontology information in upload
    - [x] Added a link to phenotypes when saving data is complete in upload.
    - [x] Fixes to chart including round measurements to the nearest integer for grouping purposes
         - This is ok for the current version but we need to change it back when making the violin plots.

ToDo for Release:
- Adding warning to download page when A) multiple traits selected and B) multiple species selected telling the user that is why the other criteria have been disabled. Use`<div class="messages warning"><p></p></div>` so that the warning is positioned correctly but still themed by Drupal.
- Validation of germplasm names should be specific to genus chosen in step 1
- hide the "Related Ontology Terms" fieldset entirely.
- Allow edit of units.

reynoldtan committed:
Enable and disable ontology fieldset

When ontology config:
1. not set - fieldset is unavailable
2. set - fieldset present with suggestions based on ontology setting
3. load ontology from reusing a trait  and has a value - fieldset
available but field is disabled
4. load ontology from reusing a trait and no value set - fieldset is
unavailable

reynoldtan committed:
Restrict germ plasm validation to project genus

reynoldtan committed:
Restrict germ plasm search to project genus

laceysanderson reviewed (commented):
ToDo for Release:
- [x] Adding warning to download page when A) multiple traits selected and B) multiple species selected telling the user that is why the other criteria have been disabled. 
- [x] Validation of germplasm names should be specific to genus chosen in step 1
- [x] Hide the "Related Ontology Terms" fieldset entirely.
- [ ] Allow edit of units.

 The first three items are approved so now we are just waiting on the editing of the units.

reynoldtan committed:
Set message to warning in configuration page

- A duplicate message for ontologies configuration set to warning
message.

reynoldtan committed:
Allow unit edit of unit field

laceysanderson reviewed (approved):
Looks good 👍 Love the "Revert" functionality!
